sql/catalyst/src/main/java/org/apache/spark/sql/connector/write/RequiresDistributionAndOrdering.java

```diff
@@ -35,6 +35,11 @@
    * Spark will distribute incoming records across partitions to satisfy the required distribution
    * before passing the records to the data source table on write.
    * <p>
+   * Batch and micro-batch writes can request a particular data distribution.
```

Comment:
This is what we discussed in PR #31700. Let me know if it's clear enough.

Reply:
We can add some validation and throw an exception for continuous execution (can be a separate PR?).

Reply:
> We can add some validation and throw an exception for continuous execution (can be a separate PR?).

+1
If it's easy to do then I'd prefer doing this in this PR. We tend to have a long time gap between PRs, even across releases.

Reply:
I agree. Let me see what can be done to add validation to continuous execution.